#!/usr/bin/env python3

# # I have neither given nor received any unauthorizedaid in completing this work,
# # nor have I presented someone else's work as my own.
# # Your Name: Devson Butani
# # LTU ID: 000732711
# # Date: 12/13/2022


PACKAGE = "term_project_pkg"  # Catkin Package Name

# Import library
from dynamic_reconfigure.parameter_generator_catkin import *

# Create Instance
gen = ParameterGenerator()

gen.add(
    "min_cup_size",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Min Pixel area (pct of image) to consider a blob as a cup",  # display on hover
    0.2,  # default
    0,  # min
    100,  # max
)

gen.add(
    "max_cup_size",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Max Pixel area (pct of image) to consider a blob as a cup",  # display on hover
    1.1,  # default
    0,  # min
    100,  # max
)

# Generate the menu in rqt tool
exit(gen.generate(PACKAGE, "detect_cup", "DetectCup"))
# 1st param - Package name constant from top
# 2nd param - node name this could be called in (for documentation only)
# 3rd param - This file's name without extension. Adds 'Config' at the end
#           - Use term_project_pkg.cfg and SnowBlowerConfig in py script

# for details: https://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
